rules/anemic-vs-rich-domain-models/rule.md

```diff
@@ -0,0 +1,111 @@
+---
+seoDescription: Understand the differences between anemic and rich domain models to improve your software architecture.
+type: rule
+title: Do you understand the difference between anemic and rich domain models?
+uri: anemic-vs-rich-domain-models
+authors:
+  - title: Daniel Mackay
+    url: https://www.ssw.com.au/people/daniel-mackay
+created: 2024-06-17T13:42:55.753Z
+guid: 346475FF-7255-4BE2-BD09-C8113CA40399
+---
+
+When using Domain-Centric architectures such as Clean Architecture, we need to decide where the business logic will go. There are two main approaches to this: Anemic Domain Model and Rich Domain Model. Understanding the differences between these two models is crucial for making informed decisions about your software architecture.
+
+<!--endintro-->
+
+## Anemic Domain Model
+
+An Anemic Domain Model is characterized by:
+
+* **Property Bags:** Entities are simple data containers with public getters and setters
+* **No Behavior:** No logic or validation within the entities
+
+### Pros of Anemic Domain Model
+
+* Good for simple or CRUD (Create, Read, Update and Delete) projects
+* Easier to understand for developers new to the project
+
+### Cons of Anemic Domain Model
+
+* Doesn’t scale well with complexity - complex logic can be duplicated across many places in the client code
+* Difficult to maintain as the project grows - changes to logic needs to be updated in multiple places
+* Less readable code - Code related to an entity is scattered across multiple places
+
+::: greybox
+class Order {
+    public int Id { get; set; }
+    public DateTime OrderDate { get; set; }
+    public decimal TotalAmount { get; set; }
+}
+:::
+Figure: Example - Anemic model where the Order class is just a data container with no behavior.
+:::
+
+## Rich Domain Model
+
+A Rich Domain Model, on the other hand, embeds business logic in the model (within Aggegates/Entities/Value Objects/Services). This approach makes the domain the heart of your system, as opposed to being database or UI-centric.
+
+A Rich Domain Model is characterized by:
+
+* **Data and Behavior:** Combines data and behavior (business logic and validation) in the same entity
+* **Encapsulation:** Entities are responsible for maintaining their own state and enforcing invariants
+
+### Pros of Rich Domain Model
+
+* Scales well with complexity - encapsulation and cohesion of Domain models (fundamental OOP principles) makes the calling client code simpler
+* Easier to maintain - logic is closer to the data it applies to
```

Comment:
Worth moving "cohesion" from the above point to this one? Your choice, feel free to ignore.

Reply:
Done